### PR TITLE
cardano-node: 1.19.1-patched -> 1.20.0

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

ADP-307 / #2098

### Overview

Updates cardano-node to [1.20.0](https://github.com/input-output-hk/cardano-node/releases/tag/1.20.0)

There are quite a few upstream API changes.

Snapshot PR is input-output-hk/cardano-haskell#28.

### CODE REVIEW

**Comment by rvl** [comment]
bors try

**Review by Anviking — APPROVED** [review]
lgtm.

Seems like swagger<->json tests are failing

File: specifications/api/swagger.yaml [hunk 1]
```diff
@@ -1398,18 +1398,59 @@ components:
         passphrase: *lenientPassphrase
         address_index: *addressIndex
 
-    TransactionMetadatum: &TransactionMetadatum
+    TransactionMetadataValue: &TransactionMetadataValue
```
Comment (KtorZ) [inline specifications/api/swagger.yaml:1401]: We need to confirm this with @AlexIOHK . In theory, 1.20.0 supports both approaches for metadata, the one we currently use with direct JSON conversion, and this one that is more of a JSON description of the metadata. 

I'd have preferred going for the JSON description initially, but we had to get partners started with something and went for the direct JSON conversion which was ready. If we change that, it'll break existing integrations (although that is rather trivially fixed).

Comment (rvl) [reply specifications/api/swagger.yaml:1401]: OK I have changed the API to use the "no schema" mapping from cardano-api. I think it's the same as before. The PR for changing to the detailed schema is #2147.

**Pull request merged**